Make November 2025 news public

cticenhour opened the pull request:
Also add template for December 2025 news

Refs #11446

@idaholab/moose-ccb Please add your contributions as well as facilitate the addition of those of your collaborators not on the CCB! Thank you!

cticenhour committed:
Make November 2025 news public

Add template for December 2025 news

Refs #11446

GiudGiud committed:
Add guillaume s november contribs

GiudGiud committed:
Add ALex Namjae Patrick Vasily Mauricio Yinbin Josh Ramiro Shikhar evil-Josh Yaqi contributions to newsletter

lindsayad committed:
Add blurb about AD array

GiudGiud reviewed (approved):
approving of content not my own

lindsayad commented on modules/doc/content/newsletter/2025/2025_11.md:
```diff
@@ -1,22 +1,87 @@
+
+The P1 radiation model was added with the linear finite volume discretization. The use of this discretization facilitates
+coupling convective flow problems, solved notably in the Navier Stokes module using this discretization, directly alongside
+radiative heat transfer.
```
```suggestion
A linear finite volume implementation of the P1 radiation model was added, and can now be used for coupled convective flow problems alongside the linear finite volume implementation of Navier-Stokes.
```

lindsayad commented on modules/doc/content/newsletter/2025/2025_11.md:
```diff
@@ -1,22 +1,87 @@
+
+A new [SCMClosure system](syntax/SCMClosures/index.md) was added to allow for user-developers to specify the closures to use
+in the subchannel solve. For now, only the friction factors can be selected in this system. Developers may create new closures
+in this system, with a dedicated base class for each physical quantity computed using a closure.
```
I think this could be made more clear. You're saying that I'd need two different base classes for computing friction factors vs. heat transfer coefficients?

GiudGiud replied:
```suggestion
for friction factors, using the dedicated [SCMFrictionClosureBase.md] base class.
Other base classes will be created for the other each physical quantity computed using closures.
```

GiudGiud commented on modules/doc/content/newsletter/2025/2025_11.md:
```diff
@@ -1,22 +1,87 @@
+  variable.
+- PETSc SNES options are no longer added by default when using the linear finite volume discretization, which does not use the nonlinear
+  SNES solver.
+- Material property output for array material properties can now support arrays of more than 10 components.
```
maybe we should rename ArrayVariables as MultiComponentVaraibles but does not roll off the tongue either

lindsayad replied:
Well array variables are perhaps most frequently represented as `RealEigenVector`, but conceptually yes we do use "array" in places for dynamically sized containers which are called `vector` in the `std`. But the name for array variables was born out of the fact that "vector" was already taken by `LAGRANGE_VEC/NEDELEC_ONE`. We probably should rename ArrayVariables.

Anyway for the sake of this line of documentation, what is the template argument to `MaterialProperty`?

lindsayad replied:
Sorry @cticenhour somehow I missed that you had modified this line in the commit you pushed yesterday. I've further updated it a bit, so this is resolved now in my opinion

GiudGiud committed:
Address Alex review

more clear

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

cticenhour commented on modules/doc/content/newsletter/2025/2025_11.md:
```diff
@@ -1,22 +1,82 @@
+
+A new [SCMClosure system](syntax/SCMClosures/index.md) was added to allow for user-developers to specify the closures to use
+in the subchannel solve. For now, only the friction factors can be selected using this system. Developers may create new closures
+for friction factors, using the dedicated [SCMFrictionClosureBase.md] base class.
```
```suggestion
for friction factors, using the dedicated SCMFrictionClosureBase base class.
```

This file does not exist.

cticenhour committed:
Fixup broken link due to missing SCM markdown file

cticenhour commented:
@lindsayad Please check in with your take on whether @GiudGiud has satisfied your review, when you have the opportunity. Thanks!

cticenhour committed:
Address suggestions from code review

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

lindsayad commented:
I just have one unresolved conversation

lindsayad committed:
Use vector language instead of array

In the code the relevant material API name is `getVectorPropertySize`. Let's be consistent with the API name as well as the fact that vectors are typically used in c++ to denote dynamically sized types while arrays can be used to denote fixed size types

cticenhour commented:
Thank you @lindsayad, @joshuahansel, and @GiudGiud!
